### DISCUSSION TITLE
Drag&Drop of tiddlers

### DISCUSSION THREAD

**Discussion by BurningTreeC** [TiddlyWiki/TiddlyWiki5#6580] [Ideas]
On https://burningtreec.github.io/tiddlywiki-multi-columns I'm experimenting with tiddler-drag-n-drop

There are some solutions I found while building it that would be feasible for the TiddlyWiki core

- `dragenabled` attribute for the draggable widget
- `dragenteractions` and `dragleaveactions` for the droppable widget
- `domNodeHeight` and `domNodeWidth` variables accessible in the `dragstartactions` action string

I'd like to discuss if we can integrate those solutions into the core

Best wishes,
Simon

**Comment by Jermolene** [comment]
Hi @BurningTreeC those all sounds reasonable – although the width/height variables should be consistent with existing widgets like the eventcatcher.

Another way of looking at it might be to explore whether tiddler drag and drop can be done with the eventcatcher widget. Things might be a little verbose, but it would give a good deal of flexibility.

If that worked, we might migrate the dropzone and draggable widgets to wikitext implementations, just as we plan for the navigator widget.

**Reply by BurningTreeC** [reply]
> Hi @BurningTreeC those all sounds reasonable – although the width/height variables should be consistent with existing widgets like the eventcatcher.
> 

Hi @Jermolene - yes those variables can also be `tv-selectednode-width` and `tv-selectednode-height` - no problem

> Another way of looking at it might be to explore whether tiddler drag and drop can be done with the eventcatcher widget. Things might be a little verbose, but it would give a good deal of flexibility.
> 

I wanted to go with an easy solution and the use of `draggable` and `droppable` widgets is the easier one

> If that worked, we might migrate the dropzone and draggable widgets to wikitext implementations, just as we plan for the navigator widget.

That sounds great, I'd like to see some work in that direction...

Best wishes,
Simon

**Comment by saqimtiaz** [comment]
@BurningTreeC also note that the draggable widget has a selector attribute now that can be used to specify what part of its contents should be used as the drag handle. This could be an alternative to toggling dragging on and off.
